Sources/SwiftRefactor/PackageManifest/AddPackageDependency.swift

```diff
@@ -0,0 +1,138 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftParser
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+/// Add a package dependency to a package manifest's source code.
+public struct AddPackageDependency: ManifestEditRefactoringProvider {
+  public struct Context {
+    public var dependency: PackageDependency
```

Comment:
I'm tempted to say this should be optional and we could add placeholder if not given?

Reply:
I'd think that the whole point of the refactoring is to actually add a dependency that's why it's non-optional at the moment, I'm not sure how we'd placeholder that efficiently because different dependency kinds have different arguments...